### PR TITLE
Migrate to ESLint 9

### PR DESCRIPTION
**Pull request by odzhychko**
With the update to v9, ESLint did not support reading configuration in subdirectories.
We relied on this feature when running ESLint with pre-commit.
Therefore, the configuration was unified at the top-level of the repository.
This resulted in a cleanup of linting related build code in the `ts-model-api` and `vue-model-api` directories.
The unification makes maintenance easier anyway.

The official ESLint hook is not used anymore because of the issue described in https://github.com/eslint/eslint/issues/18465
The local hook works around that issue.
It also does not require duplicating versions of NPM packages.
But it requires installing NPM dependencies before running pre-commit hooks.

## To be verified by reviewers

* [ ] Relevant public API members have been documented
* [ ] Documentation related to this PR is complete
  * [ ] Boundary conditions are documented
  * [ ] Exceptions are documented
  * [ ] Nullability is documented if used
* [ ] Touched existing code has been extended with documentation if missing
* [ ] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

**Comment by odzhychko** [comment]
The build fails because of the problems described here: https://github.com/eslint/eslint/issues/18465

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import pluginJs from "@eslint/js";
+import tseslint from "typescript-eslint";
+import eslintConfigPrettier from "eslint-config-prettier";
+
+export default [
+  { files: ["**/*.{js,mjs,cjs,ts}"] },
+  { ignores: ["**/.gradle", "**/dist/", "**/build/"] },
+  {
+    languageOptions: {
+      ecmaVersion: "latest",
```
Comment (languitar) [inline eslint.config.mjs:11]: Is that safe? Or might we get spurious linting error when "latest" increments?

File: eslint.config.mjs [hunk 2]
```diff
@@ -16,6 +16,13 @@ export default [
   pluginJs.configs.recommended,
   ...tseslint.configs.recommended,
   eslintConfigPrettier, // > Turns off all rules that are unnecessary or might conflict with Prettier.
+  {
+      // The Redocly plugin uses CommonJS syntax for imports and exports as in the Redocly documentation.
+      files: ["model-server-openapi/**/*"],
+      rules: {
+        "@typescript-eslint/no-require-imports": "off"
+      }
+  },
```
Comment (languitar) [inline eslint.config.mjs:25]: Maybe this change could be part of the first commit?

**Review by languitar — APPROVED** [review]
Just two thoughts while reviewing. Looks fine otherwise.

File: vue-model-api/.eslintrc.js [hunk 3]
```diff
@@ -1,44 +0,0 @@
-module.exports = {
-  env: {
-    browser: true,
-    es2021: true,
```
Comment (odzhychko) [inline vue-model-api/.eslintrc.js:4]: I missed to take the `es2021: true` to the new configuration.

File: eslint.config.mjs (hunk 1 above)
Comment (odzhychko) [reply eslint.config.mjs:11]: I just applied the value we had before.
I have to think about what effect specifying `[ecmaVersion](https://typescript-eslint.io/packages/parser/#ecmaversion)` would have.

File: eslint.config.mjs (hunk 2 above)
Comment (odzhychko) [reply eslint.config.mjs]: Done

File: vue-model-api/.eslintrc.js (hunk 3 above)
Comment (odzhychko) [reply vue-model-api/.eslintrc.js:4]: Done by using `...globals.es2021`.

File: eslint.config.mjs (hunk 1 above)
Comment (odzhychko) [reply eslint.config.mjs:11]: I do not fully understand what it does.
For me, it is out of scope to look into what effects changing the `ecmaVersion` has and to reevaluate our previous configuration of it.
Therefore, I want to keep it as it was in this PR.

**Review by languitar — APPROVED** [review]

**Pull request merged**